### PR TITLE
Fixed null city issue and renamed variables for location services

### PR DESCRIPTION
**Pull request by aarondls**
## Overview:

Functionally, the location being displayed to demonstrate the location services functionality remains the same as in the main branch. The changes here are to address the comments in the last code review.

## Solution: 

In the getCityName function (renamed to getLocationName), returning the city name is changed with returning the known name that most identifies the location. The name could be, in order of which exists first: locality, sub admin area, admin area, country name, long/lat coordinates. This avoids the issue where the city name is unknown and is set to null. Associated variable/function names with city are changed to location, where appropriate. 

Variables with snake case are also replaced with lower camel case.

## Demo:

The location services permissions request on different responses look like: 

When accepting while using app:
whileusing

When accepting location services request once
onlythistime

When denying location services request:
dontallow

Once permanently denied, the app will not ask for permission again:
permanentdeny

The app will not function until the location services permission is granted.

### CODE REVIEW

File: app/src/main/java/com/example/whattowear/MainActivity.java [hunk 1]
```diff
@@ -37,8 +37,10 @@ public class MainActivity extends AppCompatActivity implements EasyPermissions.P
     public static final int PERMISSIONS_REQUEST_CODE = 1;
 
     private FusedLocationProviderClient fusedLocationClient;
-    private Location last_location;
-    private TextView location_textview;
+    private AsyncHttpClient openweatherClient;
```
Comment (karenlulz) [inline app/src/main/java/com/example/whattowear/MainActivity.java:40]: nit: openWeatherClient

File: app/src/main/java/com/example/whattowear/MainActivity.java [hunk 2]
```diff
@@ -37,8 +37,10 @@ public class MainActivity extends AppCompatActivity implements EasyPermissions.P
...
+    private Location lastLocation;
+    private TextView locationTextview;
```
Comment (karenlulz) [inline app/src/main/java/com/example/whattowear/MainActivity.java:42]: nit: locationTextView

**Review by karenlulz — COMMENTED** [review]
Looks good! Just have some minor comments for you to address

**Comment by karenlulz** [comment]
Also regarding your demo, does you app auto close if the user denies the permissions? Or can they still interact with the app?

File: app/src/main/java/com/example/whattowear/MainActivity.java [hunk 3]
```diff
@@ -48,11 +50,15 @@ protected void onCreate(Bundle savedInstanceState) {
         // initialize fused location client
         // does not need permissions
         fusedLocationClient = LocationServices.getFusedLocationProviderClient(this);
+        openweatherClient = new AsyncHttpClient();
 
-        location_textview = findViewById(R.id.location_textview);
+        // TODO: Change this to get from Parse database
+        units = "imperial";
```
[the remark this answers is not in the document]
Comment (aarondls) [reply app/src/main/java/com/example/whattowear/MainActivity.java:56]: Sure, weatherUnits sounds good.

File: app/src/main/java/com/example/whattowear/MainActivity.java [hunk 4]
```diff
@@ -117,30 +123,29 @@ public void onPermissionsDenied(int requestCode, @NonNull List<String> perms) {
...
-                                Log.i(TAG, "Got location");
-                                Log.i(TAG, Double.toString(location.getLatitude()));
-                                Log.i(TAG, Double.toString(location.getLongitude()));
-                                String city_name = getCityName(location);
-                                Log.i(TAG, city_name);
-                                location_textview.setText(city_name);
-                                last_location = location;
+                                lastLocation = location;
+
+                                String city_name = getLocationName(location);
```
[the remark this answers is not in the document]
Comment (aarondls) [reply app/src/main/java/com/example/whattowear/MainActivity.java:143]: Sorry, yes, addressed.

File: app/src/main/java/com/example/whattowear/MainActivity.java [hunk 5]
```diff
@@ -150,48 +155,72 @@ public void onSuccess(Location location) {
...
      */
-    private String getCityName(Location location) {
+    private String getLocationName(Location location) {
         double latitude = location.getLatitude();
         double longitude = location.getLongitude();
 
         Geocoder geoCoder = new Geocoder(this, Locale.getDefault());
         StringBuilder builder = new StringBuilder();
-        String city = "";
+        String loc_name = "";
```
[the remark this answers is not in the document]
Comment (aarondls) [reply app/src/main/java/com/example/whattowear/MainActivity.java:168]: Sorry, yes, addressed.

File: app/src/main/java/com/example/whattowear/MainActivity.java [hunk 6]
```diff
@@ -150,48 +155,72 @@ public void onSuccess(Location location) {
...
             Log.e(TAG, "IOException", e);
         } catch (NullPointerException e) {
             Log.e(TAG, "NullPointerException", e);
         }
 
-        return city;
+        // TODO: Fix city being unknown
+        // if city still blank, worst case, set to long lat?
+        if (loc_name == "") {
+            loc_name = "Lat: " + latitude + "Long: " + longitude;
```
[the remark this answers is not in the document]
Comment (aarondls) [reply app/src/main/java/com/example/whattowear/MainActivity.java:191]: For now, I plan to leave it as lat/long coordinates, but if it looks ugly once I've mocked up the screen itself, then I'll try to find another workaround.

File: app/src/main/java/com/example/whattowear/MainActivity.java [hunk 7]
```diff
@@ -150,48 +155,72 @@ public void onSuccess(Location location) {
...
-        double latitude = last_location.getLatitude();
-        double longitude = last_location.getLongitude();
+        double latitude = lastLocation.getLatitude();
+        double longitude = lastLocation.getLongitude();
 
-        AsyncHttpClient client = new AsyncHttpClient();
-        String api_url = "https://api.openweathermap.org/data/3.0/onecall?lat=" + latitude + "&lon=" + longitude +"&appid="+BuildConfig.OPENWEATHER_API_KEY;
-        RequestParams params = new RequestParams();
+        // TODO: Exclude non needed data after detailed weather screen is built
+        String api_url = "https://api.openweathermap.org/data/3.0/onecall?"
```
[the remark this answers is not in the document]
Comment (aarondls) [reply app/src/main/java/com/example/whattowear/MainActivity.java:211]: Sorry, yes, addressed.

File: app/src/main/java/com/example/whattowear/MainActivity.java (hunk 1 above)
Comment (aarondls) [reply app/src/main/java/com/example/whattowear/MainActivity.java:40]: Sure

File: app/src/main/java/com/example/whattowear/MainActivity.java (hunk 2 above)
Comment (aarondls) [reply app/src/main/java/com/example/whattowear/MainActivity.java:42]: Sorry, yes, addressed.

**Comment by aarondls** [comment]
> Also regarding your demo, does you app auto close if the user denies the permissions? Or can they still interact with the app?

The app displays the warning screen always, and the user cannot interact with it until they change the permissions in the settings app. The closing part in the demo is me stopping the emulator.

**Review by karenlulz — APPROVED** [review]
Thanks for making these changes!

**Pull request merged**